work_queue/src/work_queue.c

```diff
@@ -6331,7 +6337,10 @@ struct work_queue_task *work_queue_wait_internal(struct work_queue *q, int timeo
 			// queue time statistics.
 			events++;
 			END_ACCUM_TIME(q, time_internal);
-			break;
+			if (!q->delayed_wait)
```

Comment:
Then move this if outside the  `if(t==NULL)` and change it to:  `if(t && !q->delated_wait)`